### PR TITLE
Update CPC to version 3

### PR DESCRIPTION
**Pull request by TyOverby**

### CODE REVIEW

**Comment by TyOverby** [comment]
@dotnet/roslyn-perf 

Quick review on this please.  Mostly, this updates the flags passed in to CPC which changed between version 2 and version 3.

File: src/Test/Perf/Runner/Program.cs [hunk 1]
```diff
@@ -92,18 +92,28 @@ public static void Main(string[] args)
                 int iterations;
                 if (isRunningUnderCI)
                 {
-                    Log("Running one iteration per test");
+                    if (RuntimeSettings.IsVerbose) {
+                        Log("Running one iteration per test because we are under CI");
+                    }
```
Comment (basoundr) [inline src/Test/Perf/Runner/Program.cs]: Nit: Line break

File: src/Test/Perf/Runner/Program.cs [hunk 2]
```diff
@@ -92,18 +92,28 @@ public static void Main(string[] args)
...
                     iterations = test.Iterations + 1;
+                    if (RuntimeSettings.IsVerbose) {
+                        Log("With warmup iteration");
+                    }
                 }
                 else
                 {
+                    if (RuntimeSettings.IsVerbose) {
+                        Log("No warmup iteration");
+                    }
```
Comment (basoundr) [inline src/Test/Perf/Runner/Program.cs]: same here

File: src/Test/Perf/Utilities/TraceManager.cs [hunk 3]
```diff
@@ -10,15 +10,14 @@ public class TraceManagerFactory
...
             {
-                return new TraceManager(
-                    cpcFullPath,
-                    scenarioPath);
+                Log("Found CPC, using it for trace collection");
+                return new TraceManager(cpcFullPath);
             }
             else
             {
+                Log($"WARNING: Could not find CPC at {cpcFullPath}");
```
Comment (basoundr) [inline src/Test/Perf/Utilities/TraceManager.cs]: May be add saying that there wont be any trace collected for the runs

File: src/Test/Perf/Utilities/TraceManager.cs [hunk 4]
```diff
@@ -31,31 +30,24 @@ public static ITraceManager NoOpTraceManager()
...
-            {
-                return true;
-            }
-        }
+        public bool HasWarmUpIteration => true;
 
         // Cleanup the results directory and files before every run
         public void Initialize()
         {
+            /*
```
Comment (basoundr) [inline src/Test/Perf/Utilities/TraceManager.cs]: If we dont cleanup these files then traces collected during the previous runs will become part of the current run. `ConsumptionTempResults.xml` is appended with trace info

File: src/Test/Perf/Runner/Program.cs (hunk 1 above)
Comment (TyOverby) [inline src/Test/Perf/Runner/Program.cs]: Done

File: src/Test/Perf/Runner/Program.cs (hunk 2 above)
Comment (TyOverby) [inline src/Test/Perf/Runner/Program.cs]: Done

File: src/Test/Perf/Utilities/TraceManager.cs (hunk 4 above)
Comment (TyOverby) [inline src/Test/Perf/Utilities/TraceManager.cs]: Moved to the beginning

**Comment by TyOverby** [comment]
@dotnet-bot test perf-correctness please

File: src/Test/Perf/Runner/Program.cs [hunk 5]
```diff
@@ -52,6 +52,24 @@ public static void Main(string[] args)
             }
         }
 
+        private static void Cleanup()
```
Comment (basoundr) [inline src/Test/Perf/Runner/Program.cs]: Is this called anywhere? I dont see a callsite for this method. May be I missed it

Comment (TyOverby) [inline src/Test/Perf/Runner/Program.cs]: Ack.  I pushed without commiting changes.

Comment (TyOverby) [inline src/Test/Perf/Runner/Program.cs]: Fixed

**Comment by TyOverby** [comment]
Because this PR only touches perf files, I'm going to go with the perf_correctness_prtest validation and merge this now

**Pull request merged**